src/audio/component.c

```diff
@@ -29,31 +29,68 @@ DECLARE_SOF_UUID("component", comp_uuid, 0x7c42ce8b, 0x0108, 0x43d0,
 
 DECLARE_TR_CTX(comp_tr, SOF_UUID(comp_uuid), LOG_LEVEL_INFO);
 
-static const struct comp_driver *get_drv(uint32_t type)
+static const struct comp_driver *get_drv(struct sof_ipc_comp *comp)
 {
 	struct comp_driver_list *drivers = comp_drivers_get();
 	struct list_item *clist;
 	const struct comp_driver *drv = NULL;
 	struct comp_driver_info *info;
-	uint32_t flags;
+	struct sof_ipc_comp_new_ext *comp_ext;
 
-	irq_local_disable(flags);
+	if (comp->ext_data_offset &&
+	    comp->hdr.size <
+	    comp->ext_data_offset + sizeof(struct sof_ipc_comp_new_ext)) {
+		tr_err(&comp_tr, "get_drv(): header mismatched! ext_data_offset:0x%x, hdr.size: 0x%x",
+		       comp->ext_data_offset, comp->hdr.size);
+		return NULL;
+	}
+
+	/* use UUID to match the driver if UUID is provided */
+	if (comp->ext_data_offset) {
```

Comment:
you have already checked this above.

Reply:
OK, maybe I should merge the above into this one.

Reply:
I think that code is confusing and convoluted because there's implied `comp.hdr.size-sizeof(comp_foo)` substraction. See my alternative `comp.hdr.size` suggestion at https://github.com/thesofproject/linux/pull/2215#issuecomment-648584786

When you have only additions the sanity checks become all very simple: `0 <= var <= MAX`